interfaces/cython/cantera/func1.pyx

```diff
@@ -83,7 +95,88 @@ cdef class Func1:
         return self.func.eval(t)
 
     def __reduce__(self):
-        raise NotImplementedError('Func1 object is not picklable')
+        msg = "'{}' objects are not picklable".format(type(self).__name__)
+        raise NotImplementedError(msg)
 
     def __copy__(self):
-        raise NotImplementedError('Func1 object is not copyable')
+        msg = "'{}' objects are not copyable".format(type(self).__name__)
+        raise NotImplementedError(msg)
+
+
+cdef class TabulatedFunction(Func1):
+    """
+    A `TabulatedFunction` object representing a tabulated function is defined by
+    sample points and corresponding function values. Inputs are specified either
+    by two iterable objects containing sample point location and function
+    values, or a single array that concatenates those inputs in two rows or
+    columns. Between sample points, values are evaluated based on the optional
+    keyword argument ``method``; options are ``'linear'`` (linear interpolation,
+    default) or ``'previous'`` (nearest previous value). Outside the sample
+    interval, the value at the closest end point is returned.
```

Comment:
One more change here to note the `method` is a kwarg only. I had included some text in my previous comment, but it wasn't obvious 😊

Reply:
I moved it a little further down - it's still there; the docstring mentions that `method` is a keyword argument twice.

Reply:
I think it's important to specifically clarify that it is a keyword _only_ to avoid confusion, since `*args` will take in any number of given positional arguments, leading to the error of too many arguments here.

Reply:
:thumbsup: I see what you're saying. Personally, a keyword *is* a keyword (I do not use skipped keyword names, although it's a legitimate use). I changed the wording though.